File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
  
  float money;
  int coins = 0;
  
  do {
     printf("How much change is owed?\n");
     money = get_float();
  } while (money < 0);
  
  int change = round(money * 100);
  
  while (change > 0)
	{   
	    if (change >= 25)
	    {
	        change -= 25;    
	    }
	    else if (change >= 10)
	    {
	        change -= 10;    
	    }
	    else if (change >= 5)
	    {
	        change -= 5;    
	    }
	    else
	    {
	        change -= 1;
	    }
	    
	    coins++;
	}
    
  printf("%i\n", coins);
  
}

File: pset6/greedy.py
#!/usr/bin/env python3

import cs50
import math

def main():
    
    coins = 0
    print('How much change is owed? ', end = '')
    money = cs50.get_float()
    
    while money < 0:
        print('Please provide a value greater than zero: ', end = '')
        money = cs50.get_float()
        
    change = math.floor(money * 100)
    
    while change > 0:
        
        if change >= 25:
            change = change - 25
            
        elif change >= 10:
            change = change - 10
        
        elif change >= 5:
            change = change - 5
        
        else:
            change = change -1
        
        coins = coins + 1
        
    print(coins)
            
        
    

if __name__ == "__main__":
    main()